File: epic-timeline-backend/src/main/java/com/epicstuff/model/enums/ComparisonType.java
package com.epicstuff.model.enums;

public enum ComparisonType {
    MUSICAL_INSPIRATION("Musical Inspiration"),
    HISTORICAL_EVENT("Historical Event"),
    MYTHOLOGICAL_SOURCE("Mythological Source"),
    LITERARY_REFERENCE("Literary Reference");
    
    private final String displayName;
    
    ComparisonType(String displayName) {
        this.displayName = displayName;
    }
    
    public String getDisplayName() {
        return displayName;
    }
}

File: epic-timeline-frontend/src/types/ComparisonType.ts
export enum ComparisonType {
    MUSICAL_INSPIRATION = 'MUSICAL_INSPIRATION',
    HISTORICAL_EVENT = 'HISTORICAL_EVENT',
    MYTHOLOGICAL_SOURCE = 'MYTHOLOGICAL_SOURCE',
    LITERARY_REFERENCE = 'LITERARY_REFERENCE'
}

export const ComparisonTypeDisplay: Record<ComparisonType, string> = {
    [ComparisonType.MUSICAL_INSPIRATION]: 'Musical Inspiration',
    [ComparisonType.HISTORICAL_EVENT]: 'Historical Event',
    [ComparisonType.MYTHOLOGICAL_SOURCE]: 'Mythological Source',
    [ComparisonType.LITERARY_REFERENCE]: 'Literary Reference'
};

export const ComparisonTypeColors: Record<ComparisonType, string> = {
  [ComparisonType.MUSICAL_INSPIRATION]: '#9333ea', // purple
  [ComparisonType.HISTORICAL_EVENT]: '#dc2626',    // red
  [ComparisonType.MYTHOLOGICAL_SOURCE]: '#059669', // green
  [ComparisonType.LITERARY_REFERENCE]: '#2563eb'   // blue
};

export type ComparisonTypeKey = keyof typeof ComparisonType;